### PR TITLE
[SPARK-40532][CONNECT] Add Python Version into Python UDF message

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR adds the Python version from the remote client.

See also:

https://github.com/apache/spark/blob/56c7cf33929d7d42b7d299c0bb7e895963241214/python/pyspark/context.py#L312

https://github.com/apache/spark/blob/603dc5098217d9580f611873165d25392f41cdfe/python/pyspark/worker.py#L682-L691

### Why are the changes needed?

In order to make sure to run it in the same Python version of executors.

### Does this PR introduce _any_ user-facing change?

No to end users.
Yes for the dev. Now other Python versions can be used for Spark Connect.

### How was this patch tested?

Manually tested.

### CODE REVIEW

File: python/pyspark/sql/connect/proto/commands_pb2.pyi [hunk 1]
```diff
@@ -138,9 +138,6 @@ class CreateScalarFunction(google.protobuf.message.Message):
 
     This message is used to register for example a Python UDF in the session catalog by providing
     the serialized method body.
-
-    TODO(SPARK-40532) It is required to add the interpreter / language version to the command
-      parameters.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/proto/commands_pb2.pyi:143]: This is used for `register_udf` that will be removed.

**Comment by HyukjinKwon** [comment]
cc @xinrong-meng @grundprinzip @zhengruifeng

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 2]
```diff
@@ -1447,7 +1447,7 @@ class SparkConnectPlanner(val session: SparkSession) {
       // No imported Python libraries
       Lists.newArrayList(),
       pythonExec,
-      "3.9", // TODO(SPARK-40532) This needs to be an actual Python version.
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1450]: will create a separate PR to remove these

**Review by zhengruifeng — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 3]
```diff
@@ -898,7 +898,7 @@ class SparkConnectPlanner(val session: SparkSession) {
       // No imported Python libraries
       pythonIncludes = Lists.newArrayList(),
       pythonExec = pythonExec,
-      pythonVer = "3.9", // TODO(SPARK-40532) This needs to be an actual Python version.
+      pythonVer = fun.pythonVer,
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:901]:
```suggestion
      pythonVer = fun.getPythonVer,
```

File: connector/connect/server/src/test/scala/org/apache/spark/sql/connect/messages/ConnectProtoMessagesSuite.scala [hunk 4]
```diff
@@ -81,5 +82,6 @@ class ConnectProtoMessagesSuite extends SparkFunSuite {
     assert(fun.getDeterministic == true)
     assert(fun.getArgumentsCount == 1)
     assert(fun.hasPythonUdf == true)
+    assert(fun.getPythonVer == "3.10")
```
Comment (HyukjinKwon) [inline connector/connect/server/src/test/scala/org/apache/spark/sql/connect/messages/ConnectProtoMessagesSuite.scala:85]:
```suggestion
    assert(pythonUdf.getPythonVer == "3.10")
```

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1450]: Ack!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Oh, could you resolve the conflict, @HyukjinKwon ?

**Comment by HyukjinKwon** [comment]
sure!

**Comment by dongjoon-hyun** [comment]
All tests passed. Merged to master/3.4.

**Comment by HyukjinKwon** [comment]
Thank you!

**Pull request closed without merging**